cmd/lncli/cmd_mission_control.go

```diff
@@ -85,51 +114,126 @@ func setCfg(ctx *cli.Context) error {
 
 	client := routerrpc.NewRouterClient(conn)
 
-	resp, err := client.GetMissionControlConfig(
+	// Fetch current mission control config which we update to create our
+	// response.
+	mcCfg, err := client.GetMissionControlConfig(
 		ctxc, &routerrpc.GetMissionControlConfigRequest{},
 	)
 	if err != nil {
 		return err
 	}
 
+	// haveValue is a helper variable to determine if a flag has been set or
+	// the help should be displayed.
 	var haveValue bool
 
-	if ctx.IsSet("halflife") {
-		haveValue = true
-		resp.Config.HalfLifeSeconds = uint64(ctx.Duration(
-			"halflife",
-		).Seconds())
-	}
-
-	if ctx.IsSet("hopprob") {
-		haveValue = true
-		resp.Config.HopProbability = float32(ctx.Float64("hopprob"))
-	}
-
-	if ctx.IsSet("weight") {
-		haveValue = true
-		resp.Config.Weight = float32(ctx.Float64("weight"))
-	}
-
+	// Handle general mission control settings.
 	if ctx.IsSet("pmtnr") {
 		haveValue = true
-		resp.Config.MaximumPaymentResults = uint32(ctx.Int("pmtnr"))
+		mcCfg.Config.MaximumPaymentResults = uint32(ctx.Int("pmtnr"))
 	}
-
 	if ctx.IsSet("failrelax") {
 		haveValue = true
-		resp.Config.MinimumFailureRelaxInterval = uint64(ctx.Duration(
+		mcCfg.Config.MinimumFailureRelaxInterval = uint64(ctx.Duration(
 			"failrelax",
 		).Seconds())
 	}
 
+	// We switch between estimators and set corresponding configs. If
+	// estimator is not set, we ignore the values.
+	if ctx.IsSet("estimator") {
+		switch ctx.String("estimator") {
+		case routing.AprioriEstimatorName:
+			haveValue = true
+
+			// We update all values for the apriori estimator.
+			mcCfg.Config.Model = routerrpc.
+				MissionControlConfig_APRIORI
+
+			eCfg := mcCfg.Config.EstimatorConfig
+			v, ok := eCfg.(*routerrpc.MissionControlConfig_Apriori)
+
+			// If the type assertion fails it means we are switching
+			// from a different estimator. All unset parameters are
+			// replaced with defaults by SetMissionControlConfig.
+			if !ok {
+				v = &routerrpc.MissionControlConfig_Apriori{
+					Apriori: &routerrpc.AprioriParameters{},
+				}
+			}
+
+			aCfg := v.Apriori
+			if ctx.IsSet("apriorihalflife") {
+				aCfg.HalfLifeSeconds = uint64(ctx.Duration(
+					"apriorihalflife",
+				).Seconds())
+			}
+
+			if ctx.IsSet("apriorihopprob") {
+				aCfg.HopProbability = float32(ctx.Float64(
```

Comment:
I guess all this conversions are because we do not have `ctx.Float32` in this library.

Should we use `double` instead of `float` in the proto or it always get cast on the other side too?